source/PlayerInfo.cpp

```diff
@@ -988,10 +988,18 @@ Ship *PlayerInfo::Flagship()
 // Determine which ship is the flagship and return the shared pointer to it.
 const shared_ptr<Ship> &PlayerInfo::FlagshipPtr()
 {
+	bool clearance = false;
+	if(planet)
+	{
+		for(const Mission &mission : missions)
+		{
+			clearance |= mission.HasClearance(planet);
+		}
```

Comment:
If you've found one mission with clearance, you don't need to check other missions.
```suggestion
			if(mission.HasClearance(planet))
			{
				clearance = true;
				break;
			}
```

Reply:
You live up to your username.